grype/lib.go

```diff
@@ -29,11 +29,13 @@ func FindVulnerabilities(provider vulnerability.Provider, userImageStr string, s
 		return match.Matches{}, pkg.Context{}, nil, err
 	}
 
-	return FindVulnerabilitiesForPackage(provider, context.Distro, packages...), context, packages, nil
+	matchers := matcher.NewDefaultMatchers(matcher.Config{})
```

Comment:
Maybe this `FindVulnerabilities` should be moved to the tests somehow? It looks like it's only used in tests and if anyone was using grype as a lib, it might be confusing why they can't configure java

Reply:
I don't want to use this pull request to change the current behavior of the tests. I can follow up on this comment in a second PR that refactors the tests, but I want to err on the side of less changes for now so that this change is isolated by itself without touching too many things.

Reply:
I tried moving this to the tests and the more I move this around, the more I think it's for a later PR where we make our intentions clear about grype as a library. I'd rather keep things closer to main, for this change and am grateful we still have a function available that allows our tests to validate grype behavior outside of this change with no matchers configured.

Reply:
Agree it is somewhat of a separate concern. The reason I brought it up was because this change does impact the meaning of the function to a degree. This was definitely not a blocker.